src/MCPClient/lib/clientScripts/storeFileModificationDates.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python2
+
+# This file is part of Archivematica.
+#
+# Copyright 2010-2013 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+# @package Archivematica
+# @subpackage archivematicaClientScript
+# @author Mike Cantelon <mike@artefactual.com>
+
+import datetime
+import os
+import sys
+
+import django
+django.setup()
+# dashboard
+from main import models
+# archivematicaCommon
+from custom_handlers import get_script_logger
+
+
+logger = get_script_logger('archivematica.mcp.client.storeFileModificationDates')
+
+
+def get_modification_date(shared_directory_path, transfer, transfer_file):
+    file_path_relative_to_shared_directory = transfer_file.currentlocation.replace('%transferDirectory%', transfer.currentlocation, 1)
+    file_path = file_path_relative_to_shared_directory.replace('%sharedPath%', shared_directory_path, 1)
+    mod_time = os.path.getmtime(file_path)
+    return datetime.datetime.fromtimestamp(int(mod_time)).strftime('%Y-%m-%d %H:%M:%S')
+
+
+def main(transfer_uuid, shared_directory_path):
+    transfer = models.Transfer.objects.get(uuid=transfer_uuid)
+
+    files = models.File.objects.filter(transfer=transfer)
+    for transfer_file in files:
+        transfer_file.modificationtime = get_modification_date(shared_directory_path, transfer, transfer_file)
```

Comment:
Nitpick, if you passed just a `transfer_file`, then you could access to its transfer with `transfer_file.transfer`?